CMPUT301Project/app/src/main/java/com/example/GroupProject/MainActivity.java

```diff
@@ -54,28 +54,27 @@ protected void onCreate(Bundle savedInstanceState) {
             startActivity(intent);
         }
 
-
-        ProfileFragment profileFragment = new ProfileFragment();
-        HabitFragment habitFragment = new HabitFragment();
-
         BottomNavigationView bottomNavigationView = findViewById(R.id.bottomNavigationBar);
         bottomNavigationView.setOnNavigationItemSelectedListener(item -> {
+            Fragment newFragment;
             switch (item.getItemId()) {
                 case R.id.habit:
-                    selected = R.id.habit;
-                    getSupportFragmentManager().beginTransaction().replace(R.id.nav_fragment, habitFragment).commit();
+                    newFragment = new HabitFragment();
+                    break;
+                case R.id.friends:
+                    newFragment = new FriendsFragment();
                     break;
-                case R.id.profile:
-                    selected = R.id.profile;
-                    getSupportFragmentManager().beginTransaction().replace(R.id.nav_fragment, profileFragment).commit();
+                default:
+                    newFragment = new ProfileFragment();
                     break;
             }
+            getSupportFragmentManager().beginTransaction().replace(R.id.nav_fragment, newFragment).commit();
             return true;
         });
-        if (selected == 0) {
-            bottomNavigationView.setSelectedItemId(R.id.habit);
-            selected = R.id.habit;
-        }
+
```

Comment:
Use an intent/extra to specific which fragment in MainActivity you wish to go back to